rclcpp/test/rclcpp/test_time_source.cpp

```diff
@@ -209,6 +209,8 @@ TEST_F(TestTimeSource, ROS_time_valid_attach_detach) {
 
   ts.attachNode(node);
   EXPECT_FALSE(ros_clock->ros_time_is_active());
+
+  ts.detachClock(ros_clock);
```

Comment:
Is there anything you can check to make sure detachClock worked as expected?

Reply:
I added a check similar to the rest of the test scenario here: 5d54798

I didn't see any straightforward way to test it out, the `enable_ros_time` and `disable_ros_time` could be great for that, but those are private. Let me know if you have any clue how to test this out.

Reply:
> I didn't see any straightforward way to test it out, the `enable_ros_time` and `disable_ros_time` could be great for that, but those are private. Let me know if you have any clue how to test this out.

We could consider making those protected and deriving the test class from it, but it's not required.